app/controllers/deployment_controller.rb

```diff
@@ -0,0 +1,210 @@
+class DeploymentController < ApplicationController
+  extend ActiveSupport::Concern
+
+  # before_action :check_privileges
+  after_action :cleanup_action
+
+  def new
+    @layout = 'openshift_deployment'
+    if params[:id].nil?
+      @breadcrumbs.clear
+    end
+  end
+
+  def data
+    render :json => {:data => collect_data}
+  end
+
+  def create_new_deployment
```

Comment:
@alongoldboim this part is hard to review because of the style.

Please use paragraphs and indentations to mark different parts. Also consider to split this single method into smaller ones with a meaningful names.